third_party/2and3/pytz/__init__.pyi

```diff
@@ -23,4 +23,20 @@ class _UTCclass(dt.tzinfo):
 utc = ...  # type: _UTCclass
 UTC = ...  # type: _UTCclass
 
-def timezone(zone: str) -> dt.tzinfo: ...
+
+class _BaseTzInfo(dt.tzinfo):
+    zone = ...  # type: str
+
+    def fromutc(self, dt: dt.datetime) -> dt.datetime: ...
+    def localize(self, dt: dt.datetime, is_dst: Optional[bool]=False) -> dt.datetime: ...
+
+
+class _StaticTzInfo(_BaseTzInfo):
+    def normalize(self, dt: dt.datetime, is_dst: Optional[bool]=False) -> dt.datetime: ...
+
+
+class _DstTzInfo(_BaseTzInfo):
+    def normalize(self, dt: dt.datetime) -> dt.datetime: ...
+
+
+def timezone(zone: str) -> Union[_StaticTzInfo, _DstTzInfo]: ...
```

Comment:
It's unfortunate that this returns a Union -- in many cases this will require the caller to do an isinstance() check before they can use the result further.